ee/vellum_ee/workflows/display/tests/test_base_workflow_display.py

```diff
@@ -522,3 +523,90 @@ class TestWorkflowDisplay(BaseWorkflowDisplay[TestWorkflow]):
     assert test_node is not None, "TestNode not found in serialized nodes"
     assert test_node["display_data"]["icon"] == "vellum:icon:cog"
     assert test_node["display_data"]["color"] == "navy"
+
+
+def test_base_workflow_display__graph_with_trigger_and_regular_node():
+    """
+    Tests that a workflow with both a trigger edge and a regular node edge serializes correctly.
+    """
+
+    # GIVEN a workflow with a trigger and regular nodes
+    class SlackMessageTrigger(IntegrationTrigger):
+        message: str
+
+        class Config:
+            provider = "COMPOSIO"
+            integration_name = "SLACK"
+            slug = "slack_message"
+
+    class TopNode(BaseNode):
+        class Outputs(BaseNode.Outputs):
+            result: str
+
+    class BottomNode(BaseNode):
+        class Outputs(BaseNode.Outputs):
+            final: str
+
+    class TestWorkflow(BaseWorkflow):
+        graph = {  # type: ignore[assignment]
+            TopNode,
+            SlackMessageTrigger >> BottomNode,
+        }
+
+    # WHEN we serialize the workflow
+    workflow_display = get_workflow_display(workflow_class=TestWorkflow)
+    exec_config = workflow_display.serialize()
+
+    # THEN the serialized workflow should have the expected structure
+    workflow_raw_data = exec_config["workflow_raw_data"]
+    assert isinstance(workflow_raw_data, dict)
+
+    nodes = workflow_raw_data["nodes"]
+    edges = workflow_raw_data["edges"]
+    triggers = exec_config["triggers"]
+
+    assert isinstance(nodes, list)
+    assert isinstance(edges, list)
+    assert isinstance(triggers, list)
+
+    assert len(triggers) == 1
+    trigger = triggers[0]
+    assert isinstance(trigger, dict)
+    trigger_id = trigger["id"]
+
+    entrypoint_nodes = [n for n in nodes if isinstance(n, dict) and n.get("type") == "ENTRYPOINT"]
+    assert len(entrypoint_nodes) == 1
+    entrypoint_node = entrypoint_nodes[0]
+    assert isinstance(entrypoint_node, dict)
+    entrypoint_node_id = entrypoint_node["id"]
+
+    top_node = None
+    bottom_node = None
+    for node in nodes:
```

Comment:
DevinAI use `next((node for node in nodes if ...))` to avoid nested for loop blocks in tests